File: rules/python/tests/test_idor_tp.py
# True positive: IDOR via direct object lookup without ownership check
from django.http import JsonResponse

def get_invoice(request):
    # ruleid: ez-django-idor-get-parameter
    invoice_id = request.GET.get("id")
    invoice = Invoice.objects.get(pk=invoice_id)
    return JsonResponse({"total": invoice.total})

File: rules/php/tests/test_idor_tp.php
<?php
// True positive: IDOR via direct find without scoping
class InvoiceController extends Controller
{
    public function show(Request $request)
    {
        // ruleid: ez-laravel-idor-find
        $invoice = Invoice::find($request->input('id'));
        return response()->json($invoice);
    }
}
